### ISSUE TITLE
StrawberryShake Http ResponseEnumerator breaking change 

### ISSUE THREAD

**Issue by david-driscoll** [ChilliCream/graphql-platform#5684]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Product

Strawberry Shake

### Describe the bug

There appears to be an issue introduced by #5630 that is causing strawberryshake to blow when called against a default hot chocolate server, when an error is throw by the hot chocolate graphql endpoint.

If the HC server throws an exception, and defaults to [`HttpStatusCode.InternalServerError`](https://github.com/ChilliCream/hotchocolate/blob/91e03d172a9df2905e091be6b3426b5bb1e6a514/src/HotChocolate/AspNetCore/src/AspNetCore/Serialization/DefaultHttpResponseFormatter.cs#L230-L235) then strawberry shake will throw an exception instead of [returning the correct error results](https://github.com/ChilliCream/hotchocolate/blob/91e03d172a9df2905e091be6b3426b5bb1e6a514/src/StrawberryShake/Client/src/Transport.Http/ResponseEnumerator.cs#L67-L75)

DefaultHttpResponseFormatter:
https://github.com/ChilliCream/hotchocolate/blob/91e03d172a9df2905e091be6b3426b5bb1e6a514/src/HotChocolate/AspNetCore/src/AspNetCore/Serialization/DefaultHttpResponseFormatter.cs#L230-L235

ResponseEnumerator:
https://github.com/ChilliCream/hotchocolate/blob/91e03d172a9df2905e091be6b3426b5bb1e6a514/src/StrawberryShake/Client/src/Transport.Http/ResponseEnumerator.cs#L67-L75

### Steps to reproduce

I've created a simple repro here: https://github.com/david-driscoll/hcss-http-error
If you run both unit tests `GetBook_Bad` and `GetBoth_Bad` will fail with the exception (in the logs below)

Aside: I noticed that even though the `Both` query should return the author name and the book error, [it is not currently](https://github.com/david-driscoll/hcss-http-error/blob/c8beb2785653c3fe36e363306a8c426967ed4951/tests/UnitTest1.GetBoth_Good.verified.txt). I'm not sure if this is an issue or not, just something I noticed. It could just be a configuration problem, just making a note.

### Relevant log output

```shell
System.ArgumentNullException
Either body or exception have to be set. (Parameter 'data')
   at StrawberryShake.OperationResult`1..ctor(T data, IOperationResultDataInfo dataInfo, IOperationResultDataFactory`1 dataFactory, IReadOnlyList`1 errors, IReadOnlyDictionary`2 extensions, IReadOnlyDictionary`2 contextData)
   at StrawberryShake.OperationResultBuilder`1.Build(Response`1 response)
   at StrawberryShake.StorelessOperationExecutor`2.ExecuteAsync(OperationRequest request, CancellationToken cancellationToken)
   at StrawberryShake.StorelessOperationExecutor`2.ExecuteAsync(OperationRequest request, CancellationToken cancellationToken)
   at tests.BookQuery.ExecuteAsync(CancellationToken cancellationToken) in C:\Development\temp\hc-ss-http-error\tests\obj\Debug\net7.0\berry\MyClient.Client.cs:line 656
   at tests.UnitTest1.GetBook_Bad() in C:\Development\temp\hc-ss-http-error\tests\UnitTest1.cs:line 72
   at tests.UnitTest1.GetBook_Bad() in C:\Development\temp\hc-ss-http-error\tests\UnitTest1.cs:line 73
   at Xunit.Sdk.TestInvoker`1.<>c__DisplayClass48_0.<<InvokeTestMethodAsync>b__1>d.MoveNext() in /_/src/xunit.execution/Sdk/Frameworks/Runners/TestInvoker.cs:line 264
--- End of stack trace from previous location ---
   at Xunit.Sdk.ExecutionTimer.AggregateAsync(Func`1 asyncAction) in /_/src/xunit.execution/Sdk/Frameworks/ExecutionTimer.cs:line 48
   at Xunit.Sdk.ExceptionAggregator.RunAsync(Func`1 code) in /_/src/xunit.core/Sdk/ExceptionAggregator.cs:line 90
```

### Additional Context?

I'm not sure what the correct course of action is.  This appears to have other implications if the server returns any non 200 status code.

BadRequest:
https://github.com/ChilliCream/hotchocolate/blob/91e03d172a9df2905e091be6b3426b5bb1e6a514/src/HotChocolate/AspNetCore/src/AspNetCore/Serialization/DefaultHttpResponseFormatter.cs#L207-L212

MethodNotAllowed:
https://github.com/ChilliCream/hotchocolate/blob/91e03d172a9df2905e091be6b3426b5bb1e6a514/src/HotChocolate/AspNetCore/src/AspNetCore/Serialization/DefaultHttpResponseFormatter.cs#L214-L217

Or middlewares like Authorization:
https://github.com/ChilliCream/hotchocolate/blob/91e03d172a9df2905e091be6b3426b5bb1e6a514/src/HotChocolate/Core/src/Authorization/AuthorizeValidationResultAggregator.cs#L40-L44

### Version

13.0.0-preview.96

**Comment by johan-lindqvist** [comment]
Any news on this? It's blocking us from upgrading to any of the RCs

**Comment by johan-lindqvist** [comment]
Temporary workaround after getting a tip from @PascalSenn is to add an `Accept` header with value `application/json` the HttpClient that StrawberryShake uses.

Something like this:
```csharp
serviceCollection
	.AddHttpClient("StrawberryShakeClientName")
	.ConfigureHttpClient(client =>
	{
		client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue(MediaTypeNames.Application.Json);
	});
```
Related documentation on how to add the header here: https://chillicream.com/docs/strawberryshake/v13/networking/authentication#http

**Comment by bjharo** [comment]
The workaround is not working for me, unfortunately. After adding the default header and bumping up to v13.0.5, I'm not getting the ArgumentNullException anymore but the error being returned by StrawberryShake still is not what I am expecting.

For example, I have test that submits a request to our Graphql gateway that will get back this response. The response i the same regardless if I use preview 79 or 13.0.5.

```
HTTP/1.1 500 Internal Server Error
Date: Fri, 10 Mar 2023 13:44:17 GMT
Content-Type: application/json; charset=utf-8
Transfer-Encoding: chunked
Connection: keep-alive
Server: Kestrel

{"errors":[{"message":"Cannot accept null for non-nullable input.","path":["user","user","email"],"extensions":{"code":"HC0018","field":"UserInput.email"}}]}
```

To demonstrate the issue, I debugged the test to see exactly what StrawberryShake is returned to me. 

In preview 79, I get back this, which is what I want and expect.
<img width="670" alt="Screenshot 2023-03-10 at 8 40 28 AM" src="https://user-images.githubusercontent.com/3472778/224333382-73b27a20-1897-47a1-bd0a-cfdbba2a3c72.png">

With 13.0.5, I'm getting back this, while technically correct, is not the full error being returned by the gateway.
<img width="1022" alt="Screenshot 2023-03-10 at 8 44 32 AM" src="https://user-images.githubusercontent.com/3472778/224333592-9920bf0c-da80-4011-92b1-a8a068fbaa88.png">

By chance, am I missing something that I need to do differently in v13?

**Comment by bryanh-bloomerang** [comment]
In case anyone stumbles on this issue, my problem was solved as of 13.1.0-preview.7

**Comment by michaelstaib** [comment]
Yes... this one was fixed and will be released with 13.1
